File: src/main/java/com/company/erp/features/chats/dto/ReactionDto.java
package com.company.erp.features.chats.dto;

import com.company.erp.features.chats.entity.MessageReaction;

public record ReactionDto(
        Long userId,
        String emoji
) {
    public static ReactionDto from(MessageReaction r) {
        return new ReactionDto(r.getId().getUserId(), r.getId().getEmoji());
    }
}

File: app/Features/Chat/Dto/ReactionDto.php
<?php

namespace App\Features\Chat\Dto;

use App\Features\Chat\Models\MessageReaction;

/** Port of the Spring ReactionDto record. */
final class ReactionDto
{
    /**
     * @return array<string, mixed>
     */
    public static function from(MessageReaction $r): array
    {
        return [
            'userId' => (int) $r->user_id,
            'emoji' => $r->emoji,
        ];
    }
}
